docs/proposals/04-new-core-gardener-cloud-apis.md

```diff
@@ -0,0 +1,698 @@
+    # machineImages:
+    # - name: coreos
+    #   version: 1967.5.0
+    #   publisher: CoreOS
+    #   offer: CoreOS
+    #   sku: Stable
+    # countFaultDomains:
+    # - region: westeurope
+    #   count: 2
+    # countUpdateDomains:
+    # - region: westeurope
+    #   count: 5
+    #
+    #
+    # Example for GCP:
+    #
+    # apiVersion: gcp.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2023.5.0
+    #   image: projects/coreos-cloud/global/images/coreos-stable-2023-5-0-v20190312
+    #
+    #
+    # Example for OpenStack:
+    #
+    # apiVersion: openstack.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2023.5.0
+    #   image: coreos-2023.5.0
+    # keyStoneURL: https://url-to-keystone/v3/
+    # dnsServers:
+    # - 10.10.10.10
+    # - 10.10.10.11
+    # dhcpDomain: foo.bar
+    # requestTimeout: 30s
+    # constraints:
+    #   loadBalancerProviders:
+    #   - name: haproxy
+    #   floatingPools:
+    #   - name: fip1
+    #     loadBalancerClasses:
+    #     - name: class1
+    #       floatingSubnetID: 04eed401-f85f-4610-8041-c4835c4beea6
+    #       floatingNetworkID: 23949a30-1cdd-4732-ba47-d03ced950acc
+    #       subnetID: ac46c204-9d0d-4a4c-a90d-afefe40cfc35
+    #
+    #
+    # Example for Packet:
+    #
+    # apiVersion: packet.provider.extensions.gardener.cloud/v1alpha1
+    # kind: ProviderProfileConfig
+    # machineImages:
+    # - name: coreos
+    #   version: 2079.3.0
+    #   id: d61c3912-8422-4daf-835e-854efa0062e4
+```
+
+### `Seed` resource
+
+Special note: The proposal contains field that are not yet existing in the current `garden.sapcloud.io/v1beta1.Seed` resource, but they should be implemented (open issues that require them are linked).
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: seed-secret
+  namespace: garden
+type: Opaque
+data:
+  kubeconfig: base64(kubeconfig-for-seed-cluster)
+
+---
+apiVersion: core.gardener.cloud/v1alpha1
+kind: Seed
+metadata:
+  name: seed1
+spec:
+  provider:
+    type: <some-provider-name> # {aws,azure,gcp,...}
+    region: europe-central-1
+  # Optional list of provider profiles that a shoot may use together with this seed. An empty list means that all
+  # existing provider profiles of the same provider type are supported.
+  # profiles:
+  # - providerprofile1
+  # - providerprofile2
+  secretRef:
+    name: seed-secret
+    namespace: garden
+  # Motivation for DNS section: https://github.com/gardener/gardener/issues/201.
+  dns:
+    provider: <some-provider-name> # {aws-route53, google-clouddns, ...}
+    secretName: my-dns-secret # must be in `garden` namespace
+    ingressDomain: seed1.dev.example.com
+  volume: # optional (introduced to get rid of `persistentvolume.garden.sapcloud.io/minimumSize` and `persistentvolume.garden.sapcloud.io/provider` annotations)
```

Comment:
Don't have complete idea about current use case. But with my little understanding, this is for  supporting multiple storage class (please correct me if i'm wrong). Though currently we only configure storageclass other than defaul specific to etcd-main. But could it be the case that, in future, we might want to have different storageclass for different volumes based on usage and cost tradeoff. Also, this might help us implement auto-scalability of IO parameter in future. May be i'm overthinking it (cc @amshuman-kr ). 

So, should it be array of volume configuration ?

Reply:
> Don't have complete idea about current use case. But with my little understanding, this is for supporting multiple storage class (please correct me if i'm wrong). 

Yes, it's mainly used for the storage class for etcd (or any other component in the seed).

> But could it be the case that, in future, we might want to have different storageclass for different volumes based on usage and cost tradeoff.

Is this something that is planned anytime soon? It sounds reasonable, though. Will add it.

Reply:
👍 Makes sense make this an array. Other components also might benefit by not using non-default storage class (prometheus/elastic search etc.).

Reply:
Done and also already implemented in #1308